python/src/robyn/modeling/base_model_executor.py

```diff
@@ -0,0 +1,55 @@
+# pyre-strict
+
+from abc import ABC, abstractmethod
+from typing import Any, Dict, Optional
+from robyn.data.entities.calibration_input import CalibrationInput
+from robyn.data.entities.holidays_data import HolidaysData
+from robyn.data.entities.hyperparameters import Hyperparameters
+from robyn.data.entities.mmmdata import MMMData
+from robyn.modeling.entities.enums import Models, NevergradAlgorithm
+from robyn.modeling.entities.modeloutputs import ModelOutputs
+from robyn.modeling.entities.modelrun_trials_config import TrialsConfig
+from robyn.modeling.feature_engineering import FeaturizedMMMData
+
+
+class BaseModelExecutor(ABC):
+    """
+    Abstract base class for executing a model.
+    """
+
+    def __init__(
+        self,
+        mmmdata: MMMData,
+        holidays_data: HolidaysData,
+        hyperparameters: Hyperparameters,
+        calibration_input: CalibrationInput,
+        featurized_mmm_data: FeaturizedMMMData,
+    ) -> None:
+        self.mmmdata = mmmdata
+        self.holidays_data = holidays_data
+        self.hyperparameters = hyperparameters
+        self.calibration_input = calibration_input
+        self.featurized_mmm_data = featurized_mmm_data
+
+    @abstractmethod
+    def model_run(
```

Comment:
@gufengzhou can you verify these parameters for 1. any duplicates with instance variables 2. any logical grouping that can be done for these parameters.